doc/networking.md

```diff
@@ -0,0 +1,132 @@
+The `sig` field is the broadcaster's signature using its Ethereum key, over the jobId expressed as a string. The signature consists of the resulting bytes without any further encoding.
+`sig = broadcaster.Sign(jobId.String())`
+
+Verification of `TranscoderRequest` consists of the following steps:
+1. Fetch the job based on JobId.
+    * TODO distinguish if an orchestrator is lagging behind and hasn't yet received the block with the job. Might require incorporating the node's **current** block number into the request. `CreationBlock`  as stored in the job only approximates when the job is submitted to the chain, not when it is included in a block and assigned to a transcoder.
+2. Ensure the orchestrator is assigned to the job.
+3. Ensure the current block falls within the valid range of `CreationBlock` ... `EndBlock` for the job.
+4. Verify that the signature comes from the broadcaster.
+
+The `TranscoderInfo` response contains:
+
+```protobuf
+// The orchestrator sends this in response to `GetTranscoder`, containing the
+// transcoder URI, associated credentials authorizing the broadcaster to
+// use the transcoder, and miscellaneous data related to the job.
+message TranscoderInfo {
+
+  // URI of the transcoder to use for submitting segments
+  string transcoder = 1;
+
+  // Signals the authentication method to expect within `credentials`. This
+  // field is opaque to the broadcaster, and should be passed to the transcoder.
+  string authType = 2;
+
+  // Credentials to verify the request has been authorized by an orchestrator.
+  // This field is opaque to the broadcaster.
+  string credentials = 3;
+
+  // Transcoded streamId list to update the master manifest on the broadcaster.
+  map<string, string> streamIds = 50;
+}
+```
+
+### Notes
+The authorization token will become more useful when the transcoder and orchestrator are separate entities. Additionally, the auth token can incorporate additional fields to avoid an on-chain lookup, such as the StreamID, token start time, and approximate job end time. This is stored as a base64-encoded, serialized protbuf struct.
+
+```protobuf
+// AuthToken is sent by the orchestrator and encoded in the `credentials` field
+// This record is opaque to the broadcaster and is only relevant between the
+// orchestrator and the transcoder.
+message AuthToken {
+
+  // Signature of the orchestrator over the remaining fields
+  bytes sig = 1;
+
+  int64 jobId = 10;
+}
+```
+
+## Broadcaster to Transcoder
+
+### POST `/segment`
+
+Invoked each by the broadcaster for each segment that needs to be transcoded. The transcoder address is taken from the `transcoder` field in `TranscoderInfo`.
+
+#### Required Headers:
+
+* **Authorization**
+Taken from the `authType` field in TranscoderInfo.
+* **Credentials**
+Taken from the `credentials` field in TranscoderInfo.
+* **Livepeer-Segment**
+Proves that the broadcaster generated this segment. Serialized protobuf struct, base64 encoded.
+```protobuf
+message SegData {
```

Comment:
Is the broadcaster signature going to be included in this message or is it external to this message and stored in another header field?

Reply:
Yes, the broadcaster's segment signature is in the `sig` field of SegData. In turn, that is protobuf-encoded and base64-encoded, and stored in the `Livepeer-Segment` header.

Reply:
Oops didn't see the `sig` field.